pkg/agent/clean/orphan_bindings_test.go

```diff
@@ -0,0 +1,508 @@
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "orphan binding",
+			Labels: map[string]string{
+				"ns1-non-existent-prtb": auth.PrtbInClusterBindingOwner,
+			},
+		},
+		Subjects: []v1.Subject{
+			{
+				Kind: v1.GroupKind,
+				Name: "Dogs",
+			},
+		},
+	}
+	client := orphanBindingsCleanup{
+		prtbs: &fakePRTBClient{bindings: []v3.ProjectRoleTemplateBinding{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "bad",
+				},
+			},
+		}},
+		roleBindings: &fakeRoleBindingClient{
+			bindings: []v1.RoleBinding{
+				orphanBinding,
+			},
+		},
+	}
+
+	err := client.cleanOrphans(true)
+	if err == nil {
+		t.Errorf("expected an error but didn't get it")
+	}
+
+	expectedRBs := []v1.RoleBinding{orphanBinding}
+	current, _ := client.roleBindings.List("", metav1.ListOptions{})
+	if !reflect.DeepEqual(current.Items, expectedRBs) {
+		t.Errorf("expected %v, got %v", expectedRBs, current.Items)
+	}
+}
+
+func TestCleanOrphansFailsWhenRoleBindingListingFails(t *testing.T) {
+	t.Parallel()
+	orphanBinding := v1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "bad",
+		},
+	}
+	client := orphanBindingsCleanup{
+		prtbs: &fakePRTBClient{bindings: []v3.ProjectRoleTemplateBinding{}},
+		roleBindings: &fakeRoleBindingClient{
+			bindings: []v1.RoleBinding{
+				orphanBinding,
+			},
+		},
+	}
+	if err := client.cleanOrphans(true); err == nil {
+		t.Errorf("expected an error but didn't get it")
+	}
+}
+
+func TestIsOrphanBinding(t *testing.T) {
+	t.Parallel()
+	groupSubjects := []v1.Subject{
+		{
+			Kind: v1.GroupKind,
+			Name: "Dogs",
+		},
+	}
+	client := orphanBindingsCleanup{
+		prtbHashes: map[string]struct{}{
+			"p-kmc2v_grouplonglonglonglonglonglonglonglonglonglonglong-a59f5": {},
+			"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa-4c6e1": {},
+		},
+		prtbUIDs: map[string]struct{}{
+			"c8bcebc8-779d-4fe7-a367-aa80897fc7b1": {},
+		},
+	}
+
+	tests := []struct {
+		name        string
+		binding     *v1.RoleBinding
+		isOrphan    bool
+		errExpected bool
+	}{
+		{
+			name:        "nil binding",
+			binding:     nil,
+			isOrphan:    false,
+			errExpected: false,
+		},
+		{
+			name:        "no subjects",
+			binding:     &v1.RoleBinding{},
+			isOrphan:    false,
+			errExpected: false,
+		},
+		{
+			name: "no group subjects",
+			binding: &v1.RoleBinding{
+				Subjects: []v1.Subject{
+					{
+						Kind: v1.UserKind,
+						Name: "Pug",
+					},
+				},
+			},
+			isOrphan:    false,
+			errExpected: false,
+		},
+		{
+			name: "no matching new label or uid",
+			binding: &v1.RoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"hello": "world",
+					},
+				},
+				Subjects: groupSubjects,
+			},
+			isOrphan:    false,
+			errExpected: false,
+		},
+		{
+			name: "matching new label",
+			binding: &v1.RoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"p-kmc2v_grouplonglonglonglonglonglonglonglonglonglonglong-a59f5": auth.PrtbInClusterBindingOwner,
```

Comment:
Nit: If you could include a case less then 63 chars (i.e. `p-kmc2v_c-123xyz`) that would be good, since that's what most of these will actually look like.